fs/vfs/fs_epoll.c

```diff
@@ -323,12 +368,47 @@ static int epoll_teardown(FAR epoll_head_t *eph, FAR struct epoll_event *evs,
               list_add_tail(&eph->teardown, &epn->node);
             }
         }
+      else
+        {
+          list_add_tail(&eph->teardown, &epn->node);
+        }
     }
 
   nxmutex_unlock(&eph->lock);
   return i;
 }
 
+/****************************************************************************
+ * Name: epoll_default_cb
+ *
+ * Description:
+ *   The default epoll callback function, this function do the final step of
+ *   poll notification.
+ *
+ * Input Parameters:
+ *   fds - The fds
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void epoll_default_cb(FAR struct pollfd *fds)
+{
+  FAR epoll_node_t *epn = fds->arg;
+  int semcount = 0;
+
+  epn->notified = true;
+  if (fds->revents != 0)
+    {
+      nxsem_get_value(&epn->eph->sem, &semcount);
```

Comment:
should we post until semcount == 1?

Reply:
No need, only one place call `nxsem_wait()`, the smallest value of semcount is -1.

Reply:
how about multiple callers epoll_wait?